src/gui/settings/qgssettingseditorwidgetwrapper.h

```diff
@@ -80,6 +80,20 @@ class GUI_EXPORT QgsSettingsEditorWidgetWrapper : public QObject
      */
     virtual void setWidgetFromVariant( const QVariant &value ) const = 0;
 
+    /**
+     * Enables automatic update
+     * so that the setting gets update
+     * on value change in the widget and
+     * sets the widget to the current settings value.
+     * This must called after createEditor or configureEditor.
+     * \since QGIS 3.40
```

Comment:
```suggestion
     * \warning Do NOT call this method from places where a widget is embedded in a dialog with a cancel button, as the auto-update logic will immediately overwrite the setting value and prevent rollback if the user cancels the dialog.
     *
     * \since QGIS 3.40
```